packages/state-manager/src/sagas/publicChannels/sendUnregisteredInfoMessage/sendUnregisteredInfoMessage.saga.ts

```diff
@@ -0,0 +1,34 @@
+import { put, select, call, take } from 'typed-redux-saga'
+import { messagesActions } from '../../messages/messages.slice'
+import { publicChannelsSelectors } from '../publicChannels.selectors'
+import { WriteMessagePayload, MessageType } from '@quiet/types'
+import { communitiesSelectors } from '../../communities/communities.selectors'
+import { identitySelectors } from '../../identity/identity.selectors'
+import { userJoinedMessage } from '@quiet/common'
+import { publicChannelsActions } from '../publicChannels.slice'
+
+export function* sendUnregisteredInfoMessage(): Generator {
+  const community = yield* select(communitiesSelectors.currentCommunity)
+  if (community?.CA) return
+
+  const identity = yield* select(identitySelectors.currentIdentity)
+  let generalChannel = yield* select(publicChannelsSelectors.generalChannel)
+
+  if (!generalChannel) {
+    yield* take(publicChannelsActions.channelsReplicated)
```

Comment:
How about you not only checked for the occurence of an action but also if the payload contains general channel? Then you could remove the condiiton in line 23